Mathlib/LinearAlgebra/PerfectPairing.lean

```diff
@@ -0,0 +1,56 @@
+/-
+Copyright (c) 2023 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import Mathlib.LinearAlgebra.Dual
+
+/-!
+# Perfect pairings of modules
+
+A perfect pairing of two (left) modules may be defined either as:
+ 1. A bilinear map `M × N → R` such that the induced maps `M → Dual R N` and `N → Dual R M` are both
+    bijective. (It follows from this that both `M` and `N` are both reflexive modules.)
+ 2. A linear equivalence `N ≃ Dual R M` for which `M` is reflexive. (It then follows that `N` is
+    reflexive.)
+
+The second definition is more convenient and we prove some basic facts about it here.
+
+## Main definitions
+
+ * `LinearEquiv.flip`
+ * `LinearEquiv.IsReflexive_of_equiv_Dual_of_IsReflexive`
+
+-/
+
+namespace LinearEquiv
+
+open Module
+
+variable {R M N : Type _} [CommRing R] [AddCommGroup M] [Module R M] [AddCommGroup N] [Module R N]
+variable [IsReflexive R M] (e : N ≃ₗ[R] Dual R M)
+
+/-- For a reflexive module `M`, an equivalence `N ≃ₗ[R] Dual R M` naturally yields an equivalence
+`M ≃ₗ[R] Dual R N`. Such equivalences are known as perfect pairings. -/
+noncomputable def flip : M ≃ₗ[R] Dual R N :=
```

Comment:
What do you think of `LinearEquiv.dualFlip` as the name of this lemma, for dot notation?

Reply:
I think I prefer `e.flip` for consistency / discoverability for someone who knows `LinearMap.flip`, and in a tie-break because it's got fewer characters!

Reply:
Oh, I totally missed the lemma below that explains the current name. Yes, I agree what you have now is better.